pyispyb/core/modules/userportalsync.py

```diff
@@ -0,0 +1,777 @@
+import copy
+import logging
+import time
+from typing import Any
+from datetime import datetime, timezone
+from sqlalchemy.orm import Session, joinedload
+from ispyb import models
+from pyispyb.app.extensions.database.session import engine
+from ..modules.persons import get_persons
+from ..schemas import userportalsync as schema
+from pyispyb.app.utils import timed
+from pyispyb.core.modules.utils import encode_external_id, decode_external_id
+
+
+logger = logging.getLogger("ispyb")
+
+
+def sync_proposal(proposal: schema.UserPortalProposalSync) -> time:
```

Comment:
looks like it returns `float`?

Reply:
Yep, just changed it https://github.com/ispyb/py-ispyb/pull/210/commits/fa74ea2b3eedcf652be80c64e7691bc72b08be2f